src/tools/schema2base.ts

```diff
@@ -28,8 +31,48 @@ export abstract class Schema2Base {
     }
   }
 
+  /** Collect schemas from particle connections and build map of aliases. */
+  public processManifest(manifest: Manifest): [Dictionary<Schema>, Dictionary<Set<string>>] {
+    const schemas: Dictionary<Schema> = {};
```

Comment:
You can't collect all of the schemas into a single dictionary. In C++ nested schemas need to be placed in the file before the types that use them. For example, if you have this handle:

```
in Foo {Reference<Bar {Text t}> ref} input
```

..then Bar will need to be output to the file before Foo.

That's why I had the comment below about being output first (although it does need to be updated to reflect the "nesting" idea rather than talking about being inline).

Reply:
Just ran into this problem and was going to message you about it. 

Since we have a new naming scheme, I wasn't sure if we should continue pursuing this approach -- at least in the same way. I have an almost working solution that generates aliases for the entities specified by references -- but I'm running into the ordering issue you mentioned. 

A few solutions to this problem, off the top of my head: 
- Create a new way of naming references that takes particle connection into account (change `processSchema` and `inlineSchemaName`)
- The old way (produce two dictionaries and process inline schemas first) with some modification to reflect the new naming convention
- interleave aliases with generated schemas in a way where order matters. 
- some combination of the above.

Reply:
Just pushed a fix for this issue -- I went with option 2 / reverting to the technique of collecting multiple dictionaries.